sarracen/interpolate.py

```diff
@@ -102,3 +103,105 @@ def interpolate2D(data: SarracenDataFrame,
                 image[jpix][ipix] += term * wab
 
     return image
+
+
+def interpolate2DCross(data: SarracenDataFrame,
+                       x: str,
+                       y: str,
+                       target: str,
+                       kernel: BaseKernel,
+                       x1: float = 0,
+                       y1: float = 0,
+                       x2: float = 1,
+                       y2: float = 1,
+                       pixcount: int = 150) -> ndarray:
+    """
+    Interpolates particle data in a SarracenDataFrame across two directional axes to a 1D
+    cross-section line.
+
+    :param data: The particle data, in a SarracenDataFrame.
+    :param x: The column label of the x-directional axis.
+    :param y: The column label of the y-directional axis.
+    :param target: The column label of the target smoothing data.
+    :param kernel: The kernel to use for smoothing the target data.
+    :param x1: The starting x-coordinate of the cross-section line. (in particle data space)
+    :param y1: The starting y-coordinate of the cross-section line. (in particle data space)
+    :param x2: The ending x-coordinate of the cross-section line. (in particle data space)
+    :param y2: The ending y-coordinate of the cross-section line. (in particle data space)
+    :param pixcount: The number of pixels in the output over the entire cross-sectional line.
+    :return: The interpolated output, in a 1-dimensional numpy array.
+    """
+    if np.isclose(y2, y1) and np.isclose(x2, x1):
+        raise ValueError('Zero length cross section!')
+
+    if pixcount <= 0:
+        raise ValueError('pixcount must be greater than zero!')
+
+    if kernel.ndims != 2:
+        raise ValueError("Kernel must be two-dimensional!")
+
+    output = np.zeros(pixcount)
+
+    # determine the slope of the cross-section line
+    gradient = 0
+    if not np.isclose(x2, x1):
+        gradient = (y2 - y1) / (x2 - x1)
+    yint = y2 - gradient * x2
+
+    # determine the fraction of the line that one pixel represents
+    xlength = np.sqrt((x2 - x1) ** 2 + (y2 - y1) ** 2)
+    pixwidth = xlength / pixcount
+    xpixwidth = (x2 - x1) / pixcount
+
+    # clone the necessary data columns into a new dataframe for vectorized operations
+    parts = data.copy()[['m', 'rho', 'h', target, x, y]]
```

Comment:
This will be an issue if target matches one of other columns. E.g., trying to plot the density.
Maybe this can be written without the copy()?